pkg/sql/tablewriter.go

```diff
@@ -794,6 +801,37 @@ func (td *tableDeleter) deleteAllRowsFast(
 			EndKey: tablePrefix.PrefixEnd(),
 		}
 	}
+	// If GCDeadline isn't set, assume this drop request is from a version
+	// 1.1 server and invoke legacy code that uses DeleteRange and range GC.
+	if td.tableDesc().GCDeadline == 0 {
+		return td.legacyDeleteAllRowsFast(ctx, resume, limit, traceKV)
+	}
+
+	// Assert that GC deadline is in the past.
+	if timeutil.Since(timeutil.Unix(0, td.tableDesc().GCDeadline)) < 0 {
+		panic(fmt.Sprintf("not past the GC deadline: %s", td.tableDesc()))
+	}
+	log.VEventf(ctx, 2, "ClearRange %s - %s", resume.Key, resume.EndKey)
+	crr := &roachpb.ClearRangeRequest{
+		Span: roachpb.Span{
+			Key:    resume.Key,
+			EndKey: resume.EndKey,
+		},
+		GCThreshold: hlc.Timestamp{WallTime: td.tableDesc().GCDeadline},
+	}
+	td.b.AddRawRequest(crr)
+	if _, err := td.finalize(ctx, traceKV); err != nil {
+		return resume, err
+	}
```

Comment:
Is this expected to be fast? The reason I ask is because we don't want it getting in the way of user traffic.